### DISCUSSION TITLE
How can get path pattern?

### DISCUSSION THREAD

**Discussion by wildfrontend** [vercel/next.js#50531] [App Router] [answered]
before nextjs12, i would be use `useRouter`  to get router route to do path regex . but 13 have removed it

Example like
```
const router = useRouter()
const disableRoutes =[ "/[slug]"]

const isDisable = disableRoutes.some(path => path === router.route)

```

How can i get current router`s path pattern at next app router?

**Comment by devjiwonchoi** [comment]
You can use `usePathname` to get the path pattern in `App Router`.

Example:
```
// [/blog/hello-world]

'use client';
 
import { usePathname } from 'next/navigation';
 
export default function ExampleClientComponent() {
  const pathname = usePathname();
  return <>Current pathname: {pathname}</>;  // > Current pathname: /blog/hello-world
}

```

Note that `usePathname` is Client Component and cannot be used in Server Components.
>usePathname is a [Client Component](https://nextjs.org/docs/getting-started/react-essentials) hook and is not supported in [Server Components](https://nextjs.org/docs/getting-started/react-essentials).

For more information, look [here](https://nextjs.org/docs/app/api-reference/functions/use-pathname).

**Reply by wildfrontend** [reply]
Not, `usePathname` would be return real pathname, it cant be matched whitch path i want , finally i try to use `uesParams` to check which path i need , 

routes
```
# category route need to disallow
/[category]
/[category]/detail/[id] 
/contests
/about
```

```
  const params = useParams();
const isDisallowPath = !!params?.category
``

**Reply by devjiwonchoi** [reply, marked as answer]
My bad. I misread your question, and you're right. If you want to get the [dynamic params](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes), then you need to use `useParams`. 

Note that `useParams` is also for the Client Component, so be aware that you need to add `'use client';` on the top of the file.
>useParams is a Client Component hook that lets you read a route's [dynamic params](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) filled in by the current URL.

Example:
```
'use client';
 
import { useParams } from 'next/navigation';
 
export default function ExampleClientComponent() {
  const params = useParams();
 
  // Route -> /shop/[tag]/[item]
  // URL -> /shop/shoes/nike-air-max-97
  // `params` -> { tag: 'shoes', item: 'nike-air-max-97' }
  console.log(params);
 
  return <></>;
}
```

**Comment by Uy575** [comment]
In next 13 there is hook on client side called usePathName it helps you to read the current route URL for further detail search usePathName in next docs
